### DISCUSSION TITLE
Regarding Function Hooks

### DISCUSSION THREAD

**Discussion by DevMensIT** [supabase/supabase#6215] [General]
Hello there , 
i would like to know how can i pass "HTTP Params" inside function hooks (HTTP). 
Example : 
```
{
  "list": {
    "title": "xxxx",
    "body": "xxxxxxxxxxxx"
  },
  "priority": "NORMAL",
  "data": {
    "xyz": "xxxxxxxx"
  },
  "to": [
    "lskhdfalksfhalskhdfhlasdfl",
    "lskhdfalksfhalskhdfhlasdfl"
  ]
}
```

**Comment by GaryAustin1** [comment]
A POST function will but the row contents in the body in a fixed format.
https://supabase.com/blog/2021/07/30/supabase-functions-updates

For both POST or GET you can set hard coded HTTP parameters.
![image](https://user-images.githubusercontent.com/54564956/161085545-46abcece-7f75-4d0a-a412-ea18bb9d508e.png)

**Reply by DevMensIT** [reply]
Can you share example or tutorial something to send hard-coded json to HTTP Params ?

**Comment by GaryAustin1** [comment]
I have not seen any off hand, but never looked.
If I copy in your json as is into the param value field...
I get this coming to my server:
```
?abc=%7b%20%20%20%22list%22%3a%20%7b%20%20%20%20%20%22title%22%3a%20%22xxxx%22%2c%20%20%20%20%20%22body%22%3a%20%22xxxxxxxxxxxx%22%20%20%20%7d%2c%20%20%20%22priority%22%3a%20%22NORMAL%22%2c%20%20%20%22data%22%3a%20%7b%20%20%20%20%20%22xyz%22%3a%20%22xxxxxxxx%22%20%20%20%7d%2c%20%20%20%22to%22%3a%20%5b%20%20%20%20%20%22lskhdfalksfhalskhdfhlasdfl%22%2c%20%20%20%20%20%22lskhdfalksfhalskhdfhlasdfl%22%20%20%20%5d%20%7d
```
Which converts to this with a URL converter:
```
{   "list": {     "title": "xxxx",     "body": "xxxxxxxxxxxx"   },   "priority": "NORMAL",   "data": {     "xyz": "xxxxxxxx"   },   "to": [     "lskhdfalksfhalskhdfhlasdfl",     "lskhdfalksfhalskhdfhlasdfl"   ] }
```
It is of course fixed json, you can't have it change based on row information.

**Reply by DevMensIT** [reply]
That's hard.
Do you have any sample code to call gcm api ?

**Reply by GaryAustin1** [reply]
I don't use function hooks, so would have to google if such an example exists.
I've been waiting for Supabase functions, which have just been announced today.  Then I can control exactly what I want to other APIs etc.
https://supabase.com/blog/2022/03/31/supabase-edge-functions
